engine/src/it/java/org/pentaho/di/www/RegisterPackageServletIT.java

```diff
@@ -0,0 +1,205 @@
+/*! ******************************************************************************
+ *
+ * Pentaho Data Integration
+ *
+ * Copyright (C) 2019 by Hitachi Vantara : http://www.pentaho.com
+ *
+ *******************************************************************************
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ ******************************************************************************/
+
+package org.pentaho.di.www;
+
+import org.apache.commons.io.FileUtils;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.pentaho.di.core.logging.LogChannelInterface;
+import org.pentaho.di.core.util.UUIDUtil;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.assertFalse;
+import static org.mockito.Mockito.mock;
+
+
+public class RegisterPackageServletIT {
+
+  protected RegisterPackageServlet servlet;
+
+  private static final String ZIP_FILE_NAME = "jobExport.zip";
+
+  private static final String ZIP_FILE_PATH = "src/it/resources/org/pentaho/di/www/" + ZIP_FILE_NAME;
+
+
+  @BeforeClass
+  public static void beforeAll() throws Exception
+  {
+
+    if ( getBaseTempDir().exists() ) {
+      FileUtils.cleanDirectory(getBaseTempDir());
+    }
+
+  }
+
+  @Before
+  public void setup() {
+
+    servlet = new RegisterPackageServlet();
+    LogChannelInterface logChannelInterface = mock( LogChannelInterface.class );
+    servlet.setup(new TransformationMap(),
+            new JobMap(),
+            new SocketRepository( logChannelInterface ) ,
+            new ArrayList<>() );
+  }
+
+
+  @Test
+  public void copyRequestToDirectory() throws  Exception {
+
+    File fileZip = new File(ZIP_FILE_PATH);
+    InputStream inputStream = new FileInputStream( fileZip );
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *inputStream*. Currently, there are execution paths that do not contain closure statements, for example, when RegisterPackageServletIT.createRandomTempDir throws an exception. Ensure *inputStream* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).